deeptime/src/include/deeptime/markov/msm/tram/tram.h

```diff
@@ -328,29 +369,32 @@ struct TRAM {
         std::fill(biasedConfEnergies_.mutable_data(), biasedConfEnergies_.mutable_data() +
                                                       biasedConfEnergies_.size(), inf);
 
-        auto biasMatrixBuf = input_->biasMatrixBuf();
-        auto dtrajBuf = input_->dtrajBuf();
+        std::vector<ArrayBuffer<BiasMatrix<dtype>, 2>> biasMatrixBuffers(input_->biasMatrices().begin(),
+                                                                         input_->biasMatrices().end());
+        auto biasMatrixPtr = biasMatrixBuffers.data();
 
         auto biasedConfEnergiesBuf = biasedConfEnergies_.template mutable_unchecked<2>();
         auto modifiedStateCountsLogBuf = modifiedStateCountsLog_.template unchecked<2>();
 
-        auto *scratch = scratch_.get();
+        auto nThermStates = nThermStates_, nMarkovStates = nMarkovStates_;
+	    auto input = input_;
+        #pragma omp parallel for default(none) firstprivate(nMarkovStates, nThermStates, input, biasMatrixPtr, \
+                                                            biasedConfEnergiesBuf, modifiedStateCountsLogBuf)
+        for (StateIndex i = 0; i < nMarkovStates; ++i) {
+            std::vector<dtype> scratch(nThermStates);
```

Comment:
Here it might be better to make a parallel region first in which the scratch vector is defined, then reuse it in each iteration

Reply:
its not super important, we can leave it like this